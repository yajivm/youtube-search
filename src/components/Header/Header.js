import React, { useState } from 'react';
import PropTypes from 'prop-types';
import SearchIcon from '../../assets/img/search.svg';
import Logo from '../../assets/img/logo.svg';
import UserIcon from '../../assets/img/user.svg'
import SearchBar from '../SearchBar';

import './Header.style.css';

const Header = ({ onSearch, headerRef }) => {
    const [showSearchBarForMobile, setShowSearchBarForMobile] = useState(false);

    const onClickSearchIcon = () => {
        setShowSearchBarForMobile(!showSearchBarForMobile);
    };

    return (
        <>
            <div className="header" data-testid="header">
                <div className="header-wrapper">
                    <div className="logo-wrapper">
                        <img src={Logo} alt="youtube" title="You Tube" className="logo" />
                    </div>
                    <div className="window-search-bar">
                        <SearchBar onSearch={onSearch} searchInputRef={headerRef} />
                    </div>
                    <div className="user-wrapper">
                        <button className="responsive-button search-button" data-testid="responsive-search-button" onClick={onClickSearchIcon}>
                            <img src={SearchIcon} alt="searchVideos" title="search" className="search-icon" />
                        </button>
                        <img src={UserIcon} alt="user" title="Test User" className="user-icon" />
                    </div>
                </div>
                {showSearchBarForMobile &&
                    <div className="mobile-search-bar" data-testid="mobile-search-bar">
                        <SearchBar onSearch={onSearch} data-testid="searchbar" searchInputRef={headerRef} />
                    </div>
                }
            </div>
        </>
    );
};

Header.propTypes = {
    onSearch: PropTypes.func.isRequired,
    headerRef: PropTypes.shape({}),
};

Header.defaultProps = {
    headerRef: null,
};

export default Header;
