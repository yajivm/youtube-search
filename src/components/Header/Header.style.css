.header {
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 2020;
    transform: translateY(0);
    transition: transform 0.3s ease;
    background-color: #252525;
}

.header-wrapper {
    height: 56px;
    padding: 0 16px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
}

.logo-wrapper {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.logo {
    width: 120px;
}

.user-icon {
    width: 35px;
}

.responsive-button {
    display: none;
}

.window-search-bar{
    position: relative;
    flex: 0 1 728px;
    min-width: 0px;
}