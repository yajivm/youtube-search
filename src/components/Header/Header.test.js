import React, { createRef } from 'react';
import { render, fireEvent } from '@testing-library/react';

import Header from './Header';

jest.mock('../SearchBar', () => (props) => <div data-testid="search-bar-component" {...props} />);

const mockOnSearch = jest.fn();
const mockRef = createRef();

test('should render header component', () => {
    const { getByTestId } = render(<Header onSearch={mockOnSearch} headerRef={mockRef} />);
    
    const headerElement = getByTestId('header');

    expect(headerElement).toBeDefined();
    expect(headerElement).toMatchSnapshot();
});

test('should mobile search bar when click search icon', () => {
    const { getByTestId, queryByTestId } = render(<Header onSearch={mockOnSearch} headerRef={mockRef} />);
    
    const searchIcon = getByTestId('responsive-search-button');

    expect(queryByTestId('mobile-search-bar')).toBeNull();
    expect(searchIcon).toBeDefined();

    fireEvent.click(searchIcon)

    expect(queryByTestId('mobile-search-bar')).toBeDefined();
    expect(searchIcon).toMatchSnapshot();
});