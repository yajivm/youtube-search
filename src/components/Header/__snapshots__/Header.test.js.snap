// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mobile search bar when click search icon 1`] = `
<button
  class="responsive-button search-button"
  data-testid="responsive-search-button"
>
  <img
    alt="searchVideos"
    class="search-icon"
    src="search.svg"
    title="search"
  />
</button>
`;

exports[`should render header component 1`] = `
<div
  class="header"
  data-testid="header"
>
  <div
    class="header-wrapper"
  >
    <div
      class="logo-wrapper"
    >
      <img
        alt="youtube"
        class="logo"
        src="logo.svg"
        title="You Tube"
      />
    </div>
    <div
      class="window-search-bar"
    >
      <div
        data-testid="search-bar-component"
        searchinputref="[object Object]"
      />
    </div>
    <div
      class="user-wrapper"
    >
      <button
        class="responsive-button search-button"
        data-testid="responsive-search-button"
      >
        <img
          alt="searchVideos"
          class="search-icon"
          src="search.svg"
          title="search"
        />
      </button>
      <img
        alt="user"
        class="user-icon"
        src="user.svg"
        title="Test User"
      />
    </div>
  </div>
</div>
`;
